pages/products/[id].tsx

```diff
@@ -0,0 +1,224 @@
+        main {
+          margin: 0 auto;
+          display: flex;
+          flex-direction: column;
+          align-items: center;
+        }
+        h1 {
+          font-family: 'Lucida Console', Monaco, monospace;
+          text-align: center;
+          padding: 5px;
+          text-shadow: 0px 3px 3px orange;
+        }
+        h2 {
+          font-family: 'Lucida Console', Monaco, monospace;
+          text-align: center;
+          padding: 5px;
+          background-image: url('/graybgc.jpeg');
+          text-shadow: 0px 3px 3px orange;
+        }
+        h3 {
+          font-family: 'Lucida Console', Monaco, monospace;
+          text-shadow: 0px 3px 3px orange;
+        }
+        p {
+          font-size: 15px;
+          color: darkcyan;
+          text-align: left;
+          font-weight: 700;
+          margin-top: 1em;
+          line-height: 1.6em;
+          font-size: 0.8em;
+          font-family: 'Bitter', serif;
+          max-width: 450px;
+        }
+
+        img {
+          width: 50%;
+          margin: 0 auto;
+        }
+
+        input {
+          border-radius: 5px;
+          box-shadow: 2px 3px 3px lightpink;
+          font-family: cursive;
+          font-size: 1em;
+          padding: 5px;
+          font-weight: bold;
+          margin-left: 2em;
+          width: 4em;
+          margin-left: 1em;
+        }
+        button {
+          margin-bottom: 1em;
+          width: 6em;
+          height: 3em;
+          padding: 5px;
+          border-radius: 5px;
+          cursor: pointer;
+          box-shadow: 2px 3px 3px lightpink;
+          background-color: rgb(217, 236, 230);
+          font-family: cursive;
+          font-size: 1em;
+          font-weight: bold;
+          outline: none;
+        }
+        button:hover {
+          background-color: yellow;
+          color: blue;
+          font-weight: 700;
+        }
+        button:active {
+          transition: transformY(4px);
+          background-color: rgb(235, 208, 121);
+        }
+      `}</style>
+    </div>
+  );
+};
+export default Product;
+
+// context: object => { params, req, res, query, preview, previewData}
+// params: dynamic route
+// NextPageContext: use typescript
+export async function getServerSideProps(context: NextPageContext) {
+  const id = context.query.id;
+  //const name = context.query.name;
+  // This is without using postgres, db coded manually
+  // import { getProductsById } from '../../dbFashion';
+  // const items = getProductsById(context.params.id)
+  // 1. get products (defined as items) from db which imported
+  // 2. pass items into props
+  // 3. props shows as props on top
+  // 4. by run getServerSideProps, Next.js runs this function and gives props back which is at the top
+
+  // This is the way using postgres, import function from db where require (postgres)
+  const { getProductsById } = await import('../../dbFashion.js');
+  const items = await getProductsById(id);
```

Comment:
I guess this should also be called `item` instead of `items`, since it's a single thing.